commons-component-common/src/main/java/org/exoplatform/settings/jpa/JPAUserSettingServiceImpl.java

```
@@ -0,0 +1,406 @@
+      }
+    } else {
+      saveLastReadDate(userId, 0l);
+    }
+    //
+    SettingValue<String> isEnabled = getSettingValue(userId, EXO_IS_ENABLED);
+    if (isEnabled != null) {
+      model.setEnabled(Boolean.valueOf(isEnabled.getValue()));
+    }
+    return model;
+  }
+
+  @SuppressWarnings("unchecked")
+  private SettingValue<String> getSettingValue(String userId, String propertyName) {
+    return (SettingValue<String>) settingService.get(USER.id(userId), NOTIFICATION_SCOPE, propertyName);
+  }
+
+  private List<String> getArrayListValue(String userId, String propertyName, List<String> defaultValue) {
+    SettingValue<String> values = getSettingValue(userId, propertyName);
+    if (values != null) {
+      String strs = values.getValue();
+      if ("true".equals(strs)) {
+        strs = UserSetting.EMAIL_CHANNEL;
+      }
+      if ("false".equals(strs)) {
+        return defaultValue;
+      }
+      return NotificationUtils.stringToList(getValues(strs));
+    }
+    return defaultValue;
+  }
+
+  @Override
+  public void addMixin(String userId) {
+    addMixin(new User[] { new UserImpl(userId) });
+  }
+
+  @Override
+  @ExoTransactional
+  public void addMixin(User[] users) {
+    try {
+      fillDefaultSettingsOfUser(users);
+    } catch (Exception e) {
+      LOG.error("Failed to add mixin for default setting of users", e);
+    }
+  }
+
+
+  private void removeMixin(String userId) {}
+
+  @Override
+  @ExoTransactional
+  public List<UserSetting> getUserSettingWithDeactivate() {
+    List<UserSetting> models = new ArrayList<UserSetting>();
+    try {
+      for (ContextEntity contextEntity : settingContextDAO.getContextsofType(USER.toString())) {
+        models.add(fillModel(contextEntity.getName(), settingsDAO.getUserSettingsWithDeactivate(contextEntity.getName(),
+            EXO_IS_ACTIVE, EXO_IS_ENABLED)));
+
+      }
+    } catch (Exception e) {
+      LOG.warn("Can not get the user setting with deactivated");
```

Exception is swallowed